### ISSUE TITLE
Process cmdline is truncated on Solaris

### ISSUE THREAD

**Issue by gozdal** [giampaolo/psutil#694]
`p.cmdline()` is truncated to 80 characters on Solaris 11. `/proc/PID/psinfo` that psutil uses to read the command line contains only the beginning - See e.g. https://unix.stackexchange.com/questions/3934/how-can-i-get-a-full-process-list-in-solaris-without-truncated-lines.
The solution would be to use the same method as `pargs` tool - https://github.com/illumos/illumos-gate/blob/09f79f7c66b85f056db11f58210dc6182c9b1aef/usr/src/cmd/ptools/pargs/pargs.c. It's not however clear to me how can it be easily ported as it's using `libproc`, which is documented in https://github.com/illumos/illumos-gate/blob/09f79f7c66b85f056db11f58210dc6182c9b1aef/usr/src/lib/libproc/common/libproc.h as:

```
 * In the meantime, be aware that any program linked with libproc in this
 * release of Solaris is almost guaranteed to break in the next release.
 *
 * In short, do not use this header file or libproc for any purpose.
```

As a workaround `cmdline()` could try to launch `pargs` if available but it bears a `fork` cost.

Didn't test any other Solaris version.

**Comment by giampaolo** [comment]
I don't like parsing pargs. Instead we should figure out how pargs determines the full cmdline.

**Comment by gozdal** [comment]
Me neither, but it was really a quick-fix.
I linked the source code for `pargs`, but it's 1500 lines of code which I couldn't even get to compile as it relies on `libproc` and I didn't have it installed. I guess it's possible to reimplement it.

**Comment by giampaolo** [comment]
@alxchk I think I lost track on this. Was this fixed?

**Comment by gozdal** [comment]
@alxchk Care to elaborate whether your fork fixes the problem of truncated command line on Solaris?

**Comment by gsauthof** [comment]
@gozdal , the implementation used in the fork and now included upstream should fix the truncating as the code directly reads the argv from the address-space of the running process, cf.:

https://github.com/giampaolo/psutil/blob/master/psutil/arch/solaris/environ.c#L287-L299

That means this avoids the truncating processes your user has enough permissions to look into their address space.

**Comment by gozdal** [comment]
@gsauthof O, great, I guess this issue is now solved.
